test/server/options_impl_test.cc

```diff
@@ -252,15 +252,15 @@ TEST_F(OptionsImplTest, OptionsAreInSyncWithProto) {
 TEST_F(OptionsImplTest, OptionsFromArgv) {
   const std::array<const char*, 3> args{"envoy", "-c", "hello"};
   std::unique_ptr<OptionsImpl> options = std::make_unique<OptionsImpl>(
-      args.size(), args.data(), [](bool) { return "1"; }, spdlog::level::warn);
+      static_cast<int>(args.size()), args.data(), [](bool) { return "1"; }, spdlog::level::warn);
```

Comment:
@yanavlasov Hopefully this switch from correcting the OptionsImpl API to correcting the tests that didn't reflect main(argc, argv) conventions should pass muster? Let me know if I can answer any other questions, I recognize many developers don't have (or want) a Windows machine handy to reproduce.

For completeness, this is the defect without this fix;

```
ERROR: C:/workspace/envoy/test/server/BUILD:122:1: C++ compilation of rule '//test/server:options_impl_test_lib_internal_only' failed (Exit 2)
C:\var\vcap\data\VSBuildTools\2019\VC\Tools\MSVC\14.23.28105\include\memory(2057): error C2220: the following warning is treated as an error
test/server/options_impl_test.cc(255): note: see reference to function template instantiation 'std::unique_ptr<Envoy::OptionsImpl,std::default_delete<_Ty>> std::make_unique<Envoy::OptionsImpl,std::array<const char *,3>::size_type,const char*const *,Envoy::`anonymous-namespace'::OptionsImplTest_OptionsFromArgv_Test::TestBody::<lambda_8ac950db11d983f353351aef1a781a5b>,spdlog::level::level_enum,0>(std::array<const char *,3>::size_type &&,const char *const *&&,Envoy::`anonymous-namespace'::OptionsImplTest_OptionsFromArgv_Test::TestBody::<lambda_8ac950db11d983f353351aef1a781a5b> &&,spdlog::level::level_enum &&)' being compiled
        with
        [
            _Ty=Envoy::OptionsImpl
        ]
C:\var\vcap\data\VSBuildTools\2019\VC\Tools\MSVC\14.23.28105\include\xtr1common(159): note: see reference to class template instantiation 'std::integral_constant<bool,false>' being compiled
C:\var\vcap\data\VSBuildTools\2019\VC\Tools\MSVC\14.23.28105\include\xtr1common(159): note: see reference to class template instantiation 'std::disjunction<_Traits...>' being compiled
C:\var\vcap\data\VSBuildTools\2019\VC\Tools\MSVC\14.23.28105\include\memory(2057): warning C4244: 'argument': conversion from '_Ty' to 'int', possible loss of data
        with
        [
            _Ty=unsigned __int64
        ]
```

Reply:
I see, this is where std:array is coming from. I understand now. Sorry for taking your time. I think the earlier commit where int parameter was changed to size_t is fine. It is probably cleaner than the static_cast here.

Reply:
Up to you what change you prefer.